### PR TITLE
692 support additional styles on collapsible header especially color

### LINKED ISSUE #692: Support additional styles on Collapsible header (especially color)
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**
https://discord.com/channels/1031982848485359626/1260992981322698863

### PR DESCRIPTION
**Pull request by jkorn8**
## Describe your changes
Added 3 header styles to the Collapsible widget: textColor, borderColor, and borderWidth. Also created a getHeaderStyles function that renames some of the header styles as I did not like the default names that Ant Design gave to them.

## Issue ticket number and link
[Issue #692
](https://github.com/EnsembleUI/ensemble-react/issues/692)

Closes #692 

## Checklist before requesting a review
- [ ✅] I have performed a self-review of my code
- [ ✅] I have added tests
- [ ✅] I have added a changeset `pnpm changeset add`
- [ ✅] I have added example usage in the kitchen sink app

### CODE REVIEW

File: packages/runtime/src/widgets/Collapsible.tsx [hunk 1]
```diff
@@ -41,15 +44,15 @@ interface CollapsibleContentStyles {
...
-  collpaseIcon?: IconProps;
-  expandIcon?: IconProps;
-  headerStyle?: CollapsibleHeaderStyles;
-  contentStyle?: CollapsibleContentStyles;
+  onCollapse?: EnsembleAction; // Perform an action when the Collapsible widget is collapsed
+  isAccordion?: boolean; // If true, only one panel can be expanded at a time
+  collpaseIcon?: IconProps; // The Icon displayed when the Collapsible widget is collapsed
+  expandIcon?: IconProps; // The Icon displayed when the Collapsible widget is expanded
+  headerStyle?: CollapsibleHeaderStyles; // Add one of the following 5 styles to the header: headerBg, headerPadding, textColor, borderColor, borderWidth
+  contentStyle?: CollapsibleContentStyles; // Add one of the following 2 styles to the content of the Collapsible widget: contentBg, contentPadding
```
Comment (evshi) [inline packages/runtime/src/widgets/Collapsible.tsx:55]: Can you format this as block comments similar to coreSchema.ts?

File: packages/runtime/src/widgets/Collapsible.tsx [hunk 2]
```diff
@@ -41,15 +44,15 @@ interface CollapsibleContentStyles {
...
   value: Expression<string>[];
-  onCollapse?: EnsembleAction;
-  isAccordion?: boolean;
-  collpaseIcon?: IconProps;
-  expandIcon?: IconProps;
-  headerStyle?: CollapsibleHeaderStyles;
-  contentStyle?: CollapsibleContentStyles;
+  onCollapse?: EnsembleAction; // Perform an action when the Collapsible widget is collapsed
+  isAccordion?: boolean; // If true, only one panel can be expanded at a time
+  collpaseIcon?: IconProps; // The Icon displayed when the Collapsible widget is collapsed
```
Comment (evshi) [inline packages/runtime/src/widgets/Collapsible.tsx:52]: Just noticed this typo - can you fix this?

File: packages/runtime/src/widgets/Collapsible.tsx [hunk 3]
```diff
@@ -183,12 +184,35 @@ export const Collapsible: React.FC<CollapsibleProps> = (props) => {
...
+          return { ...prev, colorTextHeading: headerStyle[key] };
+        if (key === "borderColor")
+          return { ...prev, colorBorder: headerStyle[key] };
+        if (key === "borderWidth")
+          return { ...prev, lineWidth: headerStyle[key] };
+        return {
+          ...prev,
+          [key]: headerStyle[key as keyof CollapsibleHeaderStyles],
+        };
+      },
```
Comment (evshi) [inline packages/runtime/src/widgets/Collapsible.tsx:205]: You can use lodash's `renameKeys` for this

File: .changeset/short-pens-flash.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+---
+"@ensembleui/react-kitchen-sink": minor
+"@ensembleui/react-runtime": minor
```
Comment (evshi) [inline .changeset/short-pens-flash.md:3]: Can you change these to `patch` instead of `minor`

**Review by evshi — COMMENTED** [review]
Nice work! Just a few small comments

**Comment by jkorn8** [comment]
Fixed above issues! For some reason the lodash library that was being imported did not have a renameKeys function, but I used mapKeys instead

**Review by evshi — APPROVED** [review]

**Pull request merged**